pkg/controller/eventlog/eventlog.go

```diff
@@ -0,0 +1,52 @@
+package eventlog
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ControllerWithEvents interface {
+	EventRecorder() *record.EventRecorder
+	Logger() *logr.Logger
+	Client() *client.Client
```

Comment:
I don't think this belongs as part of this interface. It's only use is in `controllerutil.LookupResource`, so I think it would be better to just pass `client client.Client` and `ControllerWithEvents` separately.

Reply:
sounds good. done